app/src/main/java/org/simple/clinic/newentry/PatientEntryScreenController.kt

```diff
@@ -217,13 +163,12 @@ class PatientEntryScreenController @Inject constructor(
         .ofType<PatientStateTextChanged>()
         .map { it.state }
 
-    val addressChanges = Observables.combineLatest(colonyOrVillageChanges, districtChanges, stateChanges)
-    { colonyOrVillage, district, state ->
-      OngoingPatientEntry.Address(colonyOrVillage.toNullable().nullIfBlank(), district, state)
-    }
+    val addressChanges = Observables
+        .combineLatest(colonyOrVillageChanges, districtChanges, stateChanges) { colonyOrVillage, district, state ->
+          Address(colonyOrVillage, district, state)
+        }
```

Comment:
Should we use function reference to make this even shorter?

```kotlin
val addressChanges = Observables.combineLatest(colonyOrVillageChanges, districtChanges, stateChanges, ::Address)
```

Reply:
Wow, I did not know this was possible. <3